src/main/java/servlets/SearchServlet.java

```diff
@@ -52,16 +58,18 @@
       "No Authentication: User must be logged in to search receipts.";
 
   private final DatastoreService datastore;
-  private final UserService userService;
+  private final UserService userService = UserServiceFactory.getUserService();
+
+  private Cursor cursor;
```

Comment:
Were you able to test that pagination works locally? If the cursor information is not passed in as part of the request how does getPage() work?

Reply:
It works both on deployment and locally. The global cursor keeps track of which page its on

Reply:
See https://wiki.sei.cmu.edu/confluence/display/java/MSC11-J.+Do+not+let+session+information+leak+within+a+servlet

Generally not a good idea to use member fields like this as it could end up leaking data from one user/client to another which is why I suggested that you pass the cursor token back to the client and have the client pass it in as a parameter.